src/diffusers/models/transformers/transformer_2d.py

```diff
@@ -506,6 +504,13 @@ def custom_forward(*inputs):
                 hidden_states = hidden_states * (1 + scale) + shift
                 hidden_states = self.proj_out(hidden_states)
                 hidden_states = hidden_states.squeeze(1)
+            else:
+                conditioning = self.transformer_blocks[0].norm1.emb(
+                    timestep, class_labels, hidden_dtype=hidden_states.dtype
+                )
+                shift, scale = self.proj_out_1(F.silu(conditioning)).chunk(2, dim=1)
+                hidden_states = self.norm_out(hidden_states) * (1 + scale[:, None]) + shift[:, None]
+                hidden_states = self.proj_out_2(hidden_states)
```

Comment:
Isn't this practically the same as what we're doing when `norm_type == "ada_norm"`?

Reply:
It is similar, but `ada_norm` doesn't take class labels as an argument. I moved this to `else` because the original was `if norm_type != ada_norm_single`. From what I can tell that block still only supports the norm used in the original DiT implementation. It might be worth it to refactor and allow other norm types.

Reply:
We can then still condition on `if class_labels is not None` or something like that no?

Reply:
If `class_labels` is `None` then you can't use `AdaLayerNormZero`. I'm not sure what the default norm should be when you want to condition on text or audio, but I picked `AdaLayerNorm` because it was similar to the zero variant without needing class labels.

Reply:
How about ada norm single i.e., the one used in PixArt Alpha?

Reply:
If you use that one without additional arguments you get this error:
```
TypeError: PixArtAlphaCombinedTimestepSizeEmbeddings(
  (time_proj): Timesteps()
  (timestep_embedder): TimestepEmbedding(
    (linear_1): Linear(in_features=256, out_features=1408, bias=True)
    (act): SiLU()
    (linear_2): Linear(in_features=1408, out_features=1408, bias=True)
  )
) argument after ** must be a mapping, not NoneType
```
It requires the arugments `resolution` and `aspect_ratio`, but those could have a default of `None` because they aren't required when `use_additional_conditions=False`

Reply:
Do you think changing that block would be more appropriate?

@yiyixuxu WDYT?